### ISSUE TITLE
Add usage example for `JsPromise` wrapper

### ISSUE THREAD

**Issue by jedel1043** [boa-dev/boa#2761]
Seeing that almost all wrappers have its own entry in `boa_examples`, it would be good to have one for the recently introduced `JsPromise` wrapper:

https://github.com/boa-dev/boa/blob/bf47815a49c71d4251bd6f8297efa8bbff03681a/boa_engine/src/object/builtins/jspromise.rs#L109-L112

**Comment by jedel1043** [comment]
Leaving this issue for newcomers. Feel free to ask for an assignment if you're interested in contributing! :D

**Comment by lvllvl** [comment]
I'd like to try my hand at this issue, if that's ok.

**Comment by jedel1043** [comment]
Let me assign it to you :)

**Comment by lvllvl** [comment]
So if I understand correctly, this would be similar to what was done between: 
- `boa_engine/src/object/builtins/jsregexp.rs` and `boa_examples/src/bin/jsregexp.rs`

Where: 
- The example in `boa_engine/src/object/builtins/jsregexp.rs` on `line 18` appears in `boa_examples/src/bin/jsregexp.rs`. 

So you'd like something similar for jspromise. Specifically: 

- Take the example in `boa_engine/src/object/builtins/jspromise.rs` (line 115-155)
- Create a file `boa_examples/src/bin/jspromise.rs` and use the example from `boa_engine/src/object/builtins/jspromise.rs` (line 115-155)

Is that correct?

**Comment by jedel1043** [comment]
Yep, essentially to "extend" the doc example of `JsPromise` with a somewhat realistic usage of the API, possibly trying to use the majority of the utility functions.

**Comment by created-by-varun** [comment]
Hi @jedel1043 I see that this PR was not merged. Is this issue available to work on?
Thanks

**Comment by jedel1043** [comment]
@created-by-varun Hello! Yeah, the last PR got stale. Since I saw you already opened up a PR for this, I'll assign it to you.

**Comment by created-by-varun** [comment]
Hi @jedel1043 
The PR is ready. Please let me know if there are any changes required.
Thanks
